### PR TITLE
#33 manage materials

### PR DESCRIPTION
**Pull request by RaidenRabit**

### CODE REVIEW

File: Code/GeorgiaLibrarySystem-/Core/App.Config [hunk 1]
```diff
@@ -6,6 +6,7 @@
   </configSections>
   <connectionStrings>
     <add name="Context" connectionString="metadata=res://*/GTLModel.csdl|res://*/GTLModel.ssdl|res://*/GTLModel.msl;provider=System.Data.SqlClient;provider connection string=&quot;data source=localhost;initial catalog=GTL;integrated security=True;MultipleActiveResultSets=True;App=EntityFramework&quot;" providerName="System.Data.EntityClient" />
+    <add name="Context" connectionString="metadata=res://*/GTLModel.csdl|res://*/GTLModel.ssdl|res://*/GTLModel.msl;provider=System.Data.SqlClient;provider connection string=&quot;data source=localhost;initial catalog=GTL;integrated security=True;multipleactiveresultsets=True;application name=EntityFramework&quot;" providerName="System.Data.EntityClient" />
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/Core/App.Config:9]: whats up with these random changes

File: Code/GeorgiaLibrarySystem-/Core/Core.csproj [hunk 2]
```diff
@@ -30,6 +30,15 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'bad test project|AnyCPU'">
+    <DebugSymbols>true</DebugSymbols>
+    <OutputPath>bin\bad test project\</OutputPath>
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/Core/Core.csproj:35]: whats up with this test project and the rest of the changes in core?

File: Code/GeorgiaLibrarySystem-/GTLService/Controller/IController/IMaterialService.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace GTLService.Controller
+{
+	
+	[ServiceContract]
+	public interface IMaterialService
+	{
+        
+        [OperationContract]
+        List<readAllMaterial> GetMaterials(string materialTitle, string author, int numOfRecords = 10, int isbn = 0, string jobStatus = "0");
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/Controller/IController/IMaterialService.cs:13]: is this for showing number of books as well, becouse if it does you have to put realy big number to be sure you see them all? what is jobStatus?

File: Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+            var materialDm = new MaterialDm_Code(materialDa_Code_Mock.Object, libraryDa_Code_Mock, personDa_Code_Mock);
+
+            //Act
+            var result = materialDm.ReadMaterials(materialTitle, author, numOfRecords, isbn, jobStatus);
+
+            //Assert
+            Assert.IsTrue(result.Count == 2);
+            Assert.IsTrue(result[0].Available_Copies == 2);
+            Assert.IsTrue(result[1].Available_Copies == 1);
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs:41]: didnt teachers say that we shouldnt have more then one assert in test?

File: Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/MaterialDa_Code.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            return _context.readAllMaterials.Find(isbn);
+        }
+
+        public virtual bool CheckMaterialIsbn(int isbn)
+        {
+            var material = _context.Materials.Find(isbn);
+            return material != null;
+        }
+
+        public virtual bool CheckCopyId(int id)
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/MaterialDa_Code.cs]: why does this material da deal with copies? shouldnt there be a new class for them (like this its harder to understand as alot of stuff in one class and leter on when expending we wont know in which class we could look for the method we need and have code duplication)

File: Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/LibraryDa_Code.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        private readonly Context _context;
+        public LibraryDa_Code(Context context)
+        {
+            this._context = context;
+        }
+
+        public virtual bool CheckLibraryName(string libraryName)
+        {
+            var library = _context.Libraries.Find(libraryName);
+            return library != null;
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/LibraryDa_Code.cs]: why is it saved it in variable which is not used

File: Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/LibraryDa_Code.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+﻿using Core;
+
+namespace GTLService.DataAccess.Code
+{
+    public class LibraryDa_Code
+    {
+        private readonly Context _context;
+        public LibraryDa_Code(Context context)
+        {
+            this._context = context;
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/LibraryDa_Code.cs]: this. is not necessary

File: Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/MaterialDa_Code.cs [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+                .Where(x => (isbn == 0 || x.ISBN.Equals(isbn.ToString())) &&
+                    (typeName.Contains("0") || x.TypeName.Contains(typeName))
+                )
+                .ToList();
+            return a;
+        }
+
+        public virtual readAllMaterial ReadMaterials(int isbn)
+        {
+            return _context.readAllMaterials.Find(isbn);
```
Comment (bubriks) [inline Code/GeorgiaLibrarySystem-/GTLService/DataAccess/Code/MaterialDa_Code.cs]: read all materials is db aproach not code

**Review by bubriks — CHANGES_REQUESTED** [review]
added some review ill look into tests more

File: Code/GeorgiaLibrarySystem-/Core/App.Config (hunk 1 above)
Comment (RaidenRabit) [reply Code/GeorgiaLibrarySystem-/Core/App.Config:9]: i have no clue m8

File: Code/GeorgiaLibrarySystem-/Core/Core.csproj (hunk 2 above)
Comment (RaidenRabit) [reply Code/GeorgiaLibrarySystem-/Core/Core.csproj:35]: idk, i got it when i merged with dev

File: Code/GeorgiaLibrarySystem-/GTLService/Controller/IController/IMaterialService.cs (hunk 3 above)
Comment (RaidenRabit) [reply Code/GeorgiaLibrarySystem-/GTLService/Controller/IController/IMaterialService.cs:13]: you can check the UI, it does show how many books are available for booking right now

File: Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs (hunk 4 above)
Comment (RaidenRabit) [reply Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs:41]: teachers said its bad to test for more than 1 thing at a time, nothing about asserts

File: Code/GeorgiaLibrarySystem-/GTLService/Controller/IController/IMaterialService.cs (hunk 3 above)
Comment (bubriks) [reply Code/GeorgiaLibrarySystem-/GTLService/Controller/IController/IMaterialService.cs:13]: ill check but what about jobStatus?

File: Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs (hunk 4 above)
Comment (bubriks) [reply Code/GeorgiaLibrarySystem-/Tests/UnitTest/MaterialTests.cs:41]: but arent you testing more then one thing at the time? i mean you check for count and the available copies

**Pull request merged**